File: Sesion 6/Marco teorico/15QUÉ OPERADORES SON POSIBLES DE SOBRECARGAR EN C++/15.py
class Numero:
    def __init__(self, valor):
        self.valor = valor

    def __add__(self, otro):
        return Numero(self.valor + otro.valor)

    def mostrar(self):
        print("Valor:", self.valor)


num1 = Numero(5)
num2 = Numero(10)

suma = num1 + num2
suma.mostrar()

File: Sesion 6/Marco teorico/15QUÉ OPERADORES SON POSIBLES DE SOBRECARGAR EN C++/15.java
class Numero {
    private int valor;

    Numero(int valor) {
        this.valor = valor;
    }

    Numero sumar(Numero otro) {
        return new Numero(this.valor + otro.valor);
    }

    void mostrar() {
        System.out.println("Valor: " + valor);
    }
}

public class Main {
    public static void main(String[] args) {
        Numero num1 = new Numero(5);
        Numero num2 = new Numero(10);

        Numero suma = num1.sumar(num2);
        suma.mostrar();
    }
}
